ArduPlane/soaring.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Plane.h"
+
+/*
+*  ArduSoar support function
+*
+*  Peter Braswell, Samuel Tabor, Andrey Kolobov, and Iain Guilliard
+*/
+void Plane::update_soaring() {
```

Comment:
I feel like this whole function should get pushed into the soaring library. Most of the function calls are to the libfrary and the only param accessed is already accessible by the library (aparm). Only thing that needs access to Plane:: is set_mode, and that can be done via callback. See how AP_Landing does it. That also allows you to push the control_mode checks into the library. Usually we don't like to do that because it makes it vehicle specific but AP_Soaring won't be on any other vehicle for the foresable future.

These are changes that can happen after the PR goes in but is something I'd like to see changed.